File: 3]. Competitive Programming/LeetCode/1]. Problems/Python/0011)_Container_With_Most_Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left=0
        right = len(height)-1
        max_area = (right-left)*min(height[right], height[left])
        
        while left < right:
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
            max_area = max(max_area, (right-left)*min(height[right], height[left]))
        return max_area

File: 3]. Competitive Programming/03]. LeetCode/1]. Problems/C++/0011)_Container_with_Most_Water.cpp
class Solution 
{
  public:
    int maxArea(vector<int>& height) 
    {
        int left = 0, right = height.size()-1;
        int max_area = (right-left)*min(height[left], height[right]);
        while(left < right)
        {
            if(height[left] < height[right])
                left++;
            else
                right--;
            max_area = max(max_area, (right-left)*min(height[left], height[right]));
        }
        return max_area;
    }
};
